autosynth/providers/nodejs-addendum.yaml

```diff
@@ -0,0 +1,25 @@
+- name: gaxios
```

Comment:
I'd like to look at other ways to approach this problem.  I am very worried about having yet another place where we have to list out all applicable targets. Two different thoughts.

**Using the GitHub Search API**
To get around this problem in `github-repo-automation`, I opted to use the GitHub Search API to identify the relevant repositories:
https://developer.github.com/v3/search/#search-repositories

The query we used there looks like this:
```
org:googleapis language:typescript language:javascript is:public archived:false
```

**Inversion of control**
Maybe a better question is, why do we need to have a centralized place for these jobs to run.  Should we consider pushing responsibility of registration down to the repository level?  I could absolutely see a future where orgs outside of googleapis (ads, analytics) want to use this tool. Is there a reason we couldn't ask repository owners to have a kokoro job on a cron that calls autosynth on their own?

Reply:
> I'd like to look at other ways to approach this problem. I am very worried about having yet another place where we have to list out all applicable targets. Two different thoughts.
> 
> **Using the GitHub Search API**
> To get around this problem in `github-repo-automation`, I opted to use the GitHub Search API to identify the relevant repositories:
> https://developer.github.com/v3/search/#search-repositories
> 
> The query we used there looks like this:
> 
> ```
> org:googleapis language:typescript language:javascript is:public archived:false
> ```

That's actually how I wrote the code the first time.  The problem is that all the queries came back marked incomplete, and indeed were missing a few results.  So autosynth would not regenerate a few unlucky repos every day.

https://developer.github.com/v3/search/#timeouts-and-incomplete-results

> **Inversion of control**
> Maybe a better question is, why do we need to have a centralized place for these jobs to run. Should we consider pushing responsibility of registration down to the repository level?

That's kinda what this code does for all but about 10 repos.  Register your new *nodejs* repo by adding synth.py to the root, and autosynth will start generating it.

If we pushed kokoro registration down to the repo levels, then we'd be maintaining 300+ kokoro job configs in piper, that have to stay in sync with 300+ build configs and synth.py files in the repos.

>I could absolutely see a future where orgs outside of googleapis (ads, analytics) want to use this tool. Is there a reason we couldn't ask repository owners to have a kokoro job on a cron that calls autosynth on their own?

I see no reason.  They can do that today.

Reply:
Ok - on the search result API thing, bummer, didn't know that, thank you :) I'm still wigged out about the 10 weird repos in the yaml.  Instead, could we use the repos list API, find all TypeScript/JavaScript APIs with a `synth.py`, and use those?
https://developer.github.com/v3/repos/#list-organization-repositories

I'm really worried we are going to create a new repo without the `nodejs-` prefix, try to start using synthtool, and forget to register it here.  Thinking out loud - I wonder if a model like this wouldn't be better to drive the entire autosynth process.  Grab every repo in the org, check if it has a synth.py, and run it.